aten/src/ATen/native/cpu/UpSampleKernel.cpp

```diff
@@ -221,6 +318,7 @@ void cpu_upsample_nearest_backward(
     at::parallel_for(0, channels, at::internal::GRAIN_SIZE / output_slice_size, loop3d);
   }
 
+  // Should this actually check for is_contiguous(input.suggest_memory_format())
```

Comment:
Please update it or create separate issue, as it is going to be extremely hard to debug why backwards on CPU doesn't preserve memory format.

Reply:
Thanks for the input @VitalyFedyunin. I will create a separate issue and link it here. Note though that there are probably other ops, I have seen in maxpool2d, where input.contiguous is call which converts to contig in channel first format if I am not mistaken, so the input format is not being preserved there as well.

Reply:
Added this issue: https://github.com/pytorch/pytorch/issues/34812